homeassistant/components/homekit_controller/climate.py

```diff
@@ -185,22 +189,24 @@ def target_temperature(self) -> float | None:
         return None
 
     @property
-    def target_temperature_step(self) -> float | None:
+    def target_temperature_step(self) -> float:
         """Return the supported step of target temperature."""
         state = self.service.value(CharacteristicsTypes.TARGET_HEATER_COOLER_STATE)
         if state == TargetHeaterCoolerStateValues.COOL and self.service.has(
             CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
         ):
-            return self.service[
-                CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
-            ].minStep
+            return (
+                self.service[CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD].minStep
+                or 1.0
+            )
         if state == TargetHeaterCoolerStateValues.HEAT and self.service.has(
             CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
         ):
-            return self.service[
-                CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
-            ].minStep
-        return None
+            return (
+                self.service[CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD].minStep
+                or 1.0
+            )
+        return 1.0
```

Comment:
Maybe you could create a `DEFAULT_MIN_STEP: Final = 1.0` constant?

Reply:
I've added one (local to homekit_controller for now). I'll do a seperate one for putting it into `.components.climate.const` if you think its worth it.